src/ChatHistoryItem.tsx

```diff
@@ -24,6 +31,10 @@ export const ChatHistoryItem = memo(
     const [editingTitle, setEditingTitle] = useState(false);
     const [editInputValue, setEditInputValue] = useState(title);
     const [isMenuOpen, setIsMenuOpen] = useState(false);
+    const [optimisticTitle, setOptimisticTitle] = useOptimistic(
+      title,
+      (_currentState, newTitle: string) => newTitle,
+    );
```

Comment:
https://react.dev/reference/react/useOptimistic

The `title` from the `activeChat` is the ultimate source of truth here. We optimistically show the newTitle, but if I were to do something like this:

```
    const [optimisticTitle, setOptimisticTitle] = useOptimistic(
      title,
      (_currentState, newTitle: string) => newTitle.concat('-pending...'),
    );
```

https://github.com/user-attachments/assets/a59b04f7-45c2-4738-ad4c-a229989869b7